File: ios/ar_charts/Sources/ar_charts/ARChartsPlugin.swift
import Flutter
import UIKit

@objc(ARChartsPlugin)
public class ARChartsPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "ar_charts", binaryMessenger: registrar.messenger())
        let instance = ARChartsPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
        let factory = LineChartViewFactory()
        registrar.register(factory, withId: "ar_charts/line_chart")
        let barFactory = BarChartViewFactory()
        registrar.register(barFactory, withId: "ar_charts/bar_chart")
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getPlatformVersion":
            result("iOS " + UIDevice.current.systemVersion)
        case "updateLineChart":
            if let args = call.arguments as? [String: Any],
               let viewIdNum = args["viewId"] as? NSNumber,
               let params = args["params"] as? [String: Any] {
                let viewId = viewIdNum.int64Value
                ChartViewRegistry.getLineChart(viewId: viewId)?.updateConfig(params: params)
            }
            result(nil)
        case "updateBarChart":
            if let args = call.arguments as? [String: Any],
               let viewIdNum = args["viewId"] as? NSNumber,
               let params = args["params"] as? [String: Any] {
                let viewId = viewIdNum.int64Value
                ChartViewRegistry.getBarChart(viewId: viewId)?.updateConfig(params: params)
            }
            result(nil)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: android/src/main/kotlin/com/anisrehman/archarts/ARChartsPlugin.kt
package com.anisrehman.archarts

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class ARChartsPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "ar_charts")
        channel.setMethodCallHandler(this)
        flutterPluginBinding
            .platformViewRegistry
            .registerViewFactory(
                "ar_charts/line_chart",
                LineChartViewFactory()
            )
        flutterPluginBinding
            .platformViewRegistry
            .registerViewFactory(
                "ar_charts/bar_chart",
                BarChartViewFactory()
            )
    }

    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
        when (call.method) {
            "getPlatformVersion" -> result.success("Android ${android.os.Build.VERSION.RELEASE}")
            "updateLineChart" -> {
                val viewId = (call.arguments as? Map<*, *>)?.get("viewId") as? Number
                val params = (call.arguments as? Map<*, *>)?.get("params") as? Map<String, Any?>
                if (viewId != null && params != null) {
                    ChartViewRegistry.getLineChart(viewId.toInt())?.updateConfig(params)
                }
                result.success(null)
            }
            "updateBarChart" -> {
                val viewId = (call.arguments as? Map<*, *>)?.get("viewId") as? Number
                val params = (call.arguments as? Map<*, *>)?.get("params") as? Map<String, Any?>
                if (viewId != null && params != null) {
                    ChartViewRegistry.getBarChart(viewId.toInt())?.updateConfig(params)
                }
                result.success(null)
            }
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }

}
